.github/workflows/build-lint-test.yml

```diff
@@ -0,0 +1,58 @@
+name: Build, Lint & Test
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+  workflow_dispatch:
+
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+        - macos-latest
```

Comment:
Do we need to build this on macos at all? It's a workflow, should only ever run on ubuntu runners, no?

Reply:
> Do we need to build this on macos at all? It's a workflow, should only ever run on ubuntu runners, no?

@kayakyakr used this for inspiration https://github.com/google-github-actions/get-secretmanager-secrets/blob/main/.github/workflows/unit.yml#L35.

I do run this on my laptop but if you tell me it's not necessary, I'll skip it.

Reply:
removed.

Reply:
It was an open question, but I don't think it's necessary, you'll be able to run these local to your computer otherwise?